### ISSUE TITLE
How to compile "raw" assets?

### ISSUE THREAD

**Issue by christianclavet** [BearishSun/BansheeEngine#93]
Hi, Been following to the letter how to compile the engine and editor, it build and compile ok.

Some assets are missing, checked and I see them in the RAW folder, but it's missing the "compiled" version with the .asset

How can I create it? Is there a guide somewhere?
Here is the crash report:

> Banshee Engine Log
> 
> System information
> 
> Banshee version: 0.4
> 
> OS version: Windows 10 64-bit
> 
> CPU vendor: GenuineIntel
> 
> CPU name: Intel(R) Core(TM) i7-3770K CPU @ 3.50GHz
> 
> CPU clock speed: 3511Mhz
> 
> CPU core count: 8
> 
> GPU #0: NVIDIA GeForce GTX 1080
> 
> GPU #1: Microsoft Basic Render Driver

> Cannot load resource. Specified file: C:\Users\Public\Projets\BansheeEngine\Data\Engine\Shaders\Transparent.bsl.asset doesn't exist.
> in class bs::TResourceHandle __cdecl bs::Resources::load(const class bs::Path &,class bs::Flags) [c:\users\public\projets\bansheeengine\source\bansheecore\source\bsresources.cpp:46]

I've created a repository from using the GIT command as mentionned on the github page, then build it successfully, then was able to compile it in debug (DX11). 

Note: Can I make it use OpenGL instead? Is the DX11 pipeline more recommended for the moment?

**Comment by BearishSun** [comment]
Assets will be compiled automatically the first time you run the engine/examples, in debug mode. Make sure to run it in debug mode and wait a bit during the first run. 

Are all the compiled resources missing or just some? It might be possible it's just some shader failing to compile, in which case you have to force asset import by deleting Timestamp.asset in Data\Engine and Data\Editor, then running the engine again, in order to get the actual compilation error.

**Comment by christianclavet** [comment]
Hi! Thanks.
I've erased the Timestamp.asset in Data\Engine (Nothing in Data\Editor), and started it again.
I had nothing appearing on screen for about a full five minutes (this time I looked at the process manager windows to see if it was still live) and the application finally appeared. I must have been trying to run it multiple times by not seeing it appear the first time...

![banshee_compiled1](https://user-images.githubusercontent.com/13395943/29250962-1d2a8d20-801a-11e7-8a81-db932d66ed69.JPG)
Got the editor window appear now, and I see new stuff in the scene, in the hierarchy window. (red). I tried to open the animation window and it crashed. (Probably because there were none)

Here is the log:

> A fatal error occurred and the program has to terminate!
> Unhandled exception at 0x140722960133039. Access violation reading 0x8.
> 
> Stack trace: 
> bs::ScriptObjectBase::getManagedInstance() - 0x140722960133039 File[bsscriptobject.h:45 (47)] Module[SBansheeEngine.dll]
> bs::ScriptComponent::internal_getComponent() - 0x140722966044588 File[bsscriptcomponent.cpp:128 (12)] Module[SBansheeEngine.dll]
> 0x2538097049738
> 0x2537478970200
> 0x2537909581168
> 0x13719456688025632768

This version was created as debug, but run directly (not from VS2015). I also remarked that it was quite slow (probably because it was using debug symbols)
